cerberus/cerberus.py

```diff
@@ -208,6 +208,37 @@ def _validate(self, document, schema=None, update=False, context=None):
         elif self.schema is None:
             raise SchemaError(errors.ERROR_SCHEMA_MISSING)
 
+
+        # if self.schema is a list, then we want to validate the document against each item
+        # in the list and return true if one schema in the list validates.
+        if isinstance( self.schema, Sequence ) and not isinstance(self.schema, _str_type):
+            # assume that we no schemas validate
+            valid = False
+            # keep an error stack to pass on in case no schemas validate
+            errorstack = []
+            # for each schema...
+            for s in self.schema:
+                validator = copy.copy(self)
+                validator.schema = s
+                validator.validate(document
+                                 , context=context
+                                 , update=update)
+                if len(validator.errors):
+                    # there were errors. append to the list
+                    errorstack.append( validator.errors )
+                else:
+                    # no errors occured, set valid and get out of here
+                    valid = True
+                    break
+
+            if not valid:
+                # no schema in the list validated.
+                # add all the errors we have collected to the errors list
+                for i in xrange(len(errorstack)):
```

Comment:
Actually, we can't. If we add errors to the `_errors` list, then we will raise an exception later on even if one of the schemas validated. We only want to add the errors if validation does not occur.